test.js

```diff
@@ -1,71 +1,528 @@
+      cache.put('key', 'value');
+      expect(cache.get('miss')).to.be.null;
+    });
+
+    it('should return the corresponding value of a key in the cache', function() {
+      cache.put('key', 'value');
+      expect(cache.get('key')).to.equal('value');
+    });
+
+    it('should handle various types of cache values', function() {
+      var values = [null, undefined, NaN, true, false, 0, 1, Number.POSITIVE_INFINITY, Number.NEGATIVE_INFINITY, '', 'a', [], {}, [1, 'a', false], {a:1,b:'a',c:false}, function() {}];
+      values.forEach(function(value, index) {
+        var key = 'key' + index;
+        cache.put(key, value);
+        expect(cache.get(key)).to.deep.equal(value);
+      });
+    });
+
+    it('should return the corresponding value of a non-expired key in the cache', function() {
+      cache.put('key', 'value', 1000);
+      clock.tick(999);
+      expect(cache.get('key')).to.equal('value');
+    });
+
+    it('should return null given an expired key', function() {
+      cache.put('key', 'value', 1000);
+      clock.tick(1000);
+      expect(cache.get('key')).to.be.null;
+    });
+
+    it('should return null given an expired key', function() {
+      cache.put('key', 'value', 1000);
+      clock.tick(1000);
+      expect(cache.get('key')).to.be.null;
+    });
+  });
+
+  describe('size()', function() {
+    before(function() {
+      cache.debug(false);
+    });
 
-cache.debug(true);
+    it('should return 0 given a fresh cache', function() {
+      expect(cache.size()).to.equal(0);
+    });
 
-var basicObject = { foo: 'b', baz: 'a', nug: 'd'};
-cache.put('obj', basicObject);
-cache.del('obj');
+    it('should return 1 after adding a single item to the cache', function() {
+      cache.put('key', 'value');
+      expect(cache.size()).to.equal(1);
+    });
 
-var notObject = "bad";
-cache.put('notObj', notObject);
-cache.del('notObj');
+    it('should return 3 after adding three items to the cache', function() {
+      cache.put('key1', 'value1');
+      cache.put('key2', 'value2');
+      cache.put('key3', 'value3');
+      expect(cache.size()).to.equal(3);
+    });
 
-cache.debug(false);
+    it('should not multi-count duplicate items added to the cache', function() {
+      cache.put('key', 'value1');
+      expect(cache.size()).to.equal(1);
+      cache.put('key', 'value2');
+      expect(cache.size()).to.equal(1);
+    });
 
-setTimeout(function() {
-  console.log('b == %s', cache.get('a'));
-}, 2000);
+    it('should update when a key in the cache expires', function() {
+      cache.put('key', 'value', 1000);
+      expect(cache.size()).to.equal(1);
+      clock.tick(999);
+      expect(cache.size()).to.equal(1);
+      clock.tick(1);
+      expect(cache.size()).to.equal(0);
+    });
+  });
 
-setTimeout(function() {
-  console.log('null == %s', cache.get('a'));
-  console.log('0 == %s', cache.size());
-}, 4000);
+  describe('memsize()', function() {
```

Comment:
I'm not convinced that `memsize()` is actually required. In fact, the tests for it are identical to the `size()` tests. I was unable to get them to ever get out of sync.